app/src/main/java/nl/nfi/cellscanner/PreferencesActivity.java

```diff
@@ -0,0 +1,294 @@
+                    a.startActivity(i);
+                    return true;
+                }
+            });
+
+            share_data_button.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+                @Override
+                public boolean onPreferenceClick(Preference preference) {
+                    a.exportData();
+                    return true;
+                }
+            });
+
+            wifi_switch.setEnabled(upload_switch.isChecked());
+
+            upload_switch.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener(){
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    wifi_switch.setEnabled((boolean)newValue);
+                    return true;
+                }
+            });
+        }
+
+        @Override
+        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+            setPreferencesFromResource(R.xml.preferences, rootKey);
+
+            a.setupRecording();
+            setupSharing();
+
+            findPreference("about_cellscanner").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+                @Override
+                public boolean onPreferenceClick(Preference preference) {
+                    AppInfoActivity.show(a);
+                    return true;
+                }
+            });
+        }
+    }
+
+    private void setupRecording() {
+        swRecordingMaster = prefs.findPreference("APP_RECORDING");
+        swGPSRecord = prefs.findPreference("GPS_RECORDING");
+        swGPSPrecision = prefs.findPreference("gps_high_accuracy");
+
+        swGPSRecord.setEnabled(swRecordingMaster.isChecked());
+        swGPSPrecision.setEnabled(swGPSRecord.isEnabled() && swGPSRecord.isChecked());
+
+        swRecordingMaster.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener(){
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newValue) {
+                boolean recording_enabled = (boolean)newValue;
+
+                if (recording_enabled)
+                    requestStartRecording();
+                else
+                    RecorderUtils.stopService(PreferencesActivity.this);
+
+                swGPSRecord.setEnabled(!recording_enabled);
+                swGPSPrecision.setEnabled(!recording_enabled && swGPSRecord.isChecked());
+
+                return true;
+            }
+        });
+
+        swGPSRecord.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener(){
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newValue) {
+                swGPSPrecision.setEnabled((boolean)newValue);
+                return true;
+            }
+        });
+
+        swGPSPrecision.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener(){
```

Comment:
Switch is no longer working on device. User is not able to toggle the switch

Reply:
possible cause: Switch is bound to GPS_HIGH_PRECISION_RECORDING in on toggle

Reply:
fixed